sdk/core/Azure.Core/src/Shared/PageResponseEnumerator.cs

```diff
@@ -11,6 +11,10 @@ namespace Azure.Core
 {
     internal static class PageResponseEnumerator
     {
+        public static Page<T> CreatePage<T>(IReadOnlyList<T> values, string continuationToken, Response response)
```

Comment:
Why not have it on Page<T>? It will be useful for Mocking scenarios and all clients won't have to add a subclass.

Reply:
Not a fan of static factories on generic types but I'll add one here.

Reply:
Most mocking framework can generates mock for abstracts also.

Reply:
Most mocking frameworks can created mock/fakes for abstract.